### PR TITLE
Exposed the CSV header map as part of CSVRecordReader

### PR DESCRIPTION
**Pull request by rajagopr**
Exposed a convenience method to retrieve the header row from the csv file.

### CODE REVIEW

File: pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java [hunk 1]
```diff
@@ -150,6 +151,10 @@ private void init()
     _iterator = _parser.iterator();
   }
 
+  public Map<String, Integer> getCSVHeaderMap() {
```
Comment (snleee) [inline pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java:154]: Can you add the javadoc comment on what's the key and what's the value for the header map?

File: pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java [hunk 2]
```diff
@@ -150,6 +151,10 @@ private void init()
...
+    return _parser.getHeaderMap();
```
Comment (snleee) [inline pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java:155]: what happens to the record reader / headermap 

if there's no header present? we can cover this in the comment?

**Review by snleee — COMMENTED** [review]
Let's add a test to cover parsing the header :)

File: pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java (hunk 1 above)
Comment (rajagopr) [reply pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java:161]: done.

File: pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java (hunk 2 above)
Comment (rajagopr) [reply pinot-plugins/pinot-input-format/pinot-csv/src/main/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReader.java:164]: added comments

**Review by snleee — APPROVED** [review]

**Pull request merged**